database/src/error.rs

```diff
@@ -64,11 +64,22 @@ pub enum InitError {
 /// There are no errors for:
 /// 1. Missing tables
 /// 2. (De)serialization
+/// 3. Shutdown errors
 ///
 /// as `cuprate_database` upholds the invariant that:
 ///
 /// 1. All tables exist
 /// 2. (De)serialization never fails
+/// 3. The database (thread-pool) only shuts down when all channels are dropped
+///
+/// TODO: Our errors + thread-pool are heavily tied in with
+/// the rest of `cuprate_database`. Using `cuprate_database` without
+/// the `service` module basically doesn't work since valid errors
+/// (like database shutting down) will panic.
```

Comment:
See `backend/heed/error.rs`. A lot of errors have knowledge of the code in `service`, e.g. `DbsFull` will just panic because _we_ know how many we use so that error should be `unreachable!()` but...

If someone uses `cuprate_database` with no relation to Monero, they could open more sub-DBs than available, in which case they would want that error to return, but instead it will just panic.

So we could add `RuntimeError::DbsFull` even though it will never happen in `service`.

Reply:
I am ok with sealing tables to prevent people from creating their own.

I don't think the fact we panic on certain errors makes cuprate_database unusable without the service, just that users will need to put more care into how they use it e.g. limit number of active txs.

This should probably be explained somewhere though.

Reply:
`Table` is now `Sealed` and only implementable on `cuprate_database::tables`.

Added some incomplete docs in `lib.rs` for the non-`service` usage.